SoomlaAndroidStore/src/com/soomla/store/data/StoreInfo.java

```diff
@@ -72,21 +76,48 @@
      * metadata when the application loads. Bumping the version is done by returning a higher number
      * in {@link IStoreAssets#getVersion()}.
      */
-    public static void setStoreAssets(IStoreAssets storeAssets) {
+
+    private static boolean assetsArrayHasMarketIdDuplicates(PurchasableVirtualItem[] assetsArray) {
+        HashSet<String> marketItemIds = new HashSet<>();
+        for (PurchasableVirtualItem pvi : assetsArray) {
+            if (pvi.getPurchaseType() instanceof PurchaseWithMarket) {
+                String currentMarketId = ((PurchaseWithMarket)pvi.getPurchaseType()).getMarketItem().getProductId();
+                if (marketItemIds.contains(currentMarketId)) {
+                    return false;
+                } else {
```

Comment:
@EugeneButusov, it's better not to use "else", if you break flow with "return" in a condition branch